`kn source ping describe foo` panics when Sink uses URI instead of Ref

### Bug report

When using `kn` to describe a PingSource, it will panic due to a nil pointer error if the Sink was defined with a URI instead of a Ref.

I believe the bug lies [in this code](https://github.com/knative/client/blob/0df54bc500ad960568e5d00285d3f112b9db665e/pkg/kn/commands/source/ping/describe.go#L93-L94):

```
	subWriter.WriteAttribute("Name", sink.Ref.Name)
	subWriter.WriteAttribute("Namespace", sink.Ref.Namespace)
```

Specifically, the code assumes that `sink.Ref` is not nil. Interestingly the code following guards for nil-ness:

```
	ref := sink.Ref
	if ref != nil {
		subWriter.WriteAttribute("Resource", fmt.Sprintf("%s (%s)", sink.Ref.Kind, sink.Ref.APIVersion))
	}
```

### Expected behavior

I expect that `kn source ping describe` does not have an error when inspecting a Sink that uses a URI rather than a Ref.

### Steps to reproduce the problem

```
$ kn source ping create example --sink https://example.com/sink
Ping source 'example' created in namespace 'default'.

$ kn source ping describe ping-player
Name:         example
Namespace:    default
Annotations:  [redacted]
Age:          2m
Schedule:     * * * * *
Data:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1fa81b9]

goroutine 1 [running]:
knative.dev/client/pkg/kn/commands/source/ping.writeSink(0x24f8c60, 0xc0005b6b70, 0xc00026cc78)
	/home/prow/go/src/knative.dev/client/pkg/kn/commands/source/ping/describe.go:93 +0x89
knative.dev/client/pkg/kn/commands/source/ping.NewPingDescribeCommand.func1(0xc00024cdc0, 0xc0003f61a0, 0x1, 0x1, 0x0, 0x0)
	/home/prow/go/src/knative.dev/client/pkg/kn/commands/source/ping/describe.go:69 +0x1f1
github.com/spf13/cobra.(*Command).execute(0xc00024cdc0, 0xc0003f6170, 0x1, 0x1, 0xc00024cdc0, 0xc0003f6170)
	/home/prow/go/src/knative.dev/client/vendor/github.com/spf13/cobra/command.go:832 +0x460
github.com/spf13/cobra.(*Command).ExecuteC(0xc0001722c0, 0x161043a4a23d1638, 0x3049d20, 0xc00001df10)
	/home/prow/go/src/knative.dev/client/vendor/github.com/spf13/cobra/command.go:920 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
	/home/prow/go/src/knative.dev/client/vendor/github.com/spf13/cobra/command.go:870
main.main()
	/home/prow/go/src/knative.dev/client/cmd/kn/main.go:36 +0xc8
```

### kn version

```
Version:      v0.14.0
Build Date:   2020-04-22 08:56:32
Git Revision: c41e9fd
Supported APIs:
* Serving
  - serving.knative.dev/v1 (knative-serving v0.14.0)
* Eventing
  - sources.knative.dev/v1alpha2 (knative-eventing v0.14.1)
  - eventing.knative.dev/v1alpha1 (knative-eventing v0.14.1)
```

### Knative (serving/eventing) version

v0.14.0.

--- a/CHANGELOG.adoc
+++ b/CHANGELOG.adoc
@@ -16,6 +16,10 @@
 |===
 | | Description | PR
 
+| 🐛
+| Fix Panic for kn source ping describe with Sink URI
+| https://github.com/knative/client/pull/848[#848]
+
 | 🎁
 | Add kn service delete --all
 | https://github.com/knative/client/pull/836[#836]

--- a/pkg/kn/commands/source/ping/describe.go
+++ b/pkg/kn/commands/source/ping/describe.go
@@ -90,10 +90,10 @@ func NewPingDescribeCommand(p *commands.KnParams) *cobra.Command {
 
 func writeSink(dw printers.PrefixWriter, sink *duckv1.Destination) {
 	subWriter := dw.WriteAttribute("Sink", "")
-	subWriter.WriteAttribute("Name", sink.Ref.Name)
-	subWriter.WriteAttribute("Namespace", sink.Ref.Namespace)
 	ref := sink.Ref
 	if ref != nil {
+		subWriter.WriteAttribute("Name", sink.Ref.Name)
+		subWriter.WriteAttribute("Namespace", sink.Ref.Namespace)
 		subWriter.WriteAttribute("Resource", fmt.Sprintf("%s (%s)", sink.Ref.Kind, sink.Ref.APIVersion))
 	}
 	uri := sink.URI

--- a/pkg/kn/commands/source/ping/describe_test.go
+++ b/pkg/kn/commands/source/ping/describe_test.go
@@ -25,21 +25,33 @@ import (
 
 	clientv1alpha2 "knative.dev/client/pkg/sources/v1alpha2"
 	"knative.dev/client/pkg/util"
+	"knative.dev/pkg/apis"
 )
 
-func TestSimpleDescribe(t *testing.T) {
+func TestDescribeRef(t *testing.T) {
 	pingClient := clientv1alpha2.NewMockKnPingSourceClient(t, "mynamespace")
 
 	pingRecorder := pingClient.Recorder()
-	pingRecorder.GetPingSource("testsource", getPingSource(), nil)
+	pingRecorder.GetPingSource("testsource-ref", getPingSourceSinkRef(), nil)
 
-	out, err := executePingSourceCommand(pingClient, nil, "describe", "testsource")
+	out, err := executePingSourceCommand(pingClient, nil, "describe", "testsource-ref")
 	assert.NilError(t, err)
-	util.ContainsAll(out, "1 2 3 4 5", "honeymoon", "myservicenamespace", "mysvc", "Service", "testsource")
-	util.ContainsAll(out, "myaccount", "100m", "128Mi", "200m", "256Mi")
+	assert.Assert(t, util.ContainsAll(out, "1 2 3 4 5", "honeymoon", "myservicenamespace", "mysvc", "Service", "testsource-ref"))
 
 	pingRecorder.Validate()
+}
 
+func TestDescribeURI(t *testing.T) {
+	pingClient := clientv1alpha2.NewMockKnPingSourceClient(t, "mynamespace")
+
+	pingRecorder := pingClient.Recorder()
+	pingRecorder.GetPingSource("testsource-uri", getPingSourceSinkURI(), nil)
+
+	out, err := executePingSourceCommand(pingClient, nil, "describe", "testsource-uri")
+	assert.NilError(t, err)
+	assert.Assert(t, util.ContainsAll(out, "mynamespace", "1 2 3 4 5", "honeymoon", "URI", "https", "foo", "testsource-uri"))
+
+	pingRecorder.Validate()
 }
 
 func TestDescribeError(t *testing.T) {
@@ -50,17 +62,17 @@ func TestDescribeError(t *testing.T) {
 
 	out, err := executePingSourceCommand(pingClient, nil, "describe", "testsource")
 	assert.ErrorContains(t, err, "testsource")
-	util.ContainsAll(out, "Usage", "testsource")
+	assert.Assert(t, util.ContainsAll(out, "Usage", "testsource"))
 
 	pingRecorder.Validate()
 
 }
 
-func getPingSource() *v1alpha2.PingSource {
+func getPingSourceSinkRef() *v1alpha2.PingSource {
 	return &v1alpha2.PingSource{
 		TypeMeta: metav1.TypeMeta{},
 		ObjectMeta: metav1.ObjectMeta{
-			Name: "testsource",
+			Name: "testsource-ref",
 		},
 		Spec: v1alpha2.PingSourceSpec{
 			Schedule: "1 2 3 4 5",
@@ -71,6 +83,29 @@ func getPingSource() *v1alpha2.PingSource {
 						Kind:      "Service",
 						Namespace: "myservicenamespace",
 						Name:      "mysvc",
+					},
+				},
+			},
+		},
+		Status: v1alpha2.PingSourceStatus{},
+	}
+}
+
+func getPingSourceSinkURI() *v1alpha2.PingSource {
+	return &v1alpha2.PingSource{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "testsource-uri",
+			Namespace: "mynamespace",
+		},
+		Spec: v1alpha2.PingSourceSpec{
+			Schedule: "1 2 3 4 5",
+			JsonData: "honeymoon",
+			SourceSpec: duckv1.SourceSpec{
+				Sink: duckv1.Destination{
+					URI: &apis.URL{
+						Scheme: "https",
+						Host:   "foo",
 					},
 				},
 			},

